File: src/p1005/python/solution.py
from typing import List
import math


class Solution:
    def largestSumAfterKNegations(self, nums: List[int], k: int) -> int:
        nums.sort()
        total, value = 0, math.inf
        for num in nums:
            if num < 0 < k:
                num, k = -num, k - 1
            total += num
            value = min(value, num)
        return total - (k % 2) * 2 * value

File: src/p1005/java/solution.java
import java.util.Arrays;

class Solution {
    public int largestSumAfterKNegations(int[] nums, int k) {
        Arrays.sort(nums);
        int total = 0, value = Integer.MAX_VALUE;
        for (int num : nums) {
            if (num < 0 && k > 0) {
                num = -num; k--;
            }
            total += num;
            value = Math.min(value, num);
        }
        return total - (k % 2) * 2 * value;
    }
}
